resources/assets/components/CampaignIdSingle/index.js

```diff
@@ -15,13 +17,70 @@ class CampaignIdSingle extends React.Component {
     super(props);
 
     this.state = {
+      actions: {},
       displayCreateActionModal: false,
     };
 
+    this.api = new RogueClient(window.location.origin, {
+      headers: {
+        Authorization: `Bearer ${window.AUTH}`,
+      },
+    });
+
+    this.deleteAction = this.deleteAction.bind(this);
     // this.showCreateActionModal = this.showCreateActionModal.bind(this);
     // this.hideCreateActionModal = this.hideCreateActionModal.bind(this);
   }
 
+  componentDidMount() {
+    this.getActions(this.props.campaign.id);
+  }
+
+  /**
+   * Gets the campaign's actions.
+   *
+   * @param {Integer} campaignId
+   * @return {Object}
+   */
+  getActions(campaignId) {
+    this.api
+      .get(`api/v3/actions`, {
+        filter: {
+          campaign_id: campaignId,
+        },
+      })
+      .then(json =>
+        this.setState({
+          actions: keyBy(json.data, 'id'),
+        }),
+      );
+  }
+
+  // Delete an action.
+  deleteAction(actionId, event) {
+    event.preventDefault();
+    const confirmed = confirm(
+      '🚨🔥🚨 Are you sure you want to delete this? 🚨🔥🚨',
```

Comment:
Excellent choice of emojis!! Could be clearer to add the word "action" at the end maybe!

Reply:
Ah I just copy and pasted that from what we had already when deleting a post - I think maybe @sbsmith86 or @DFurnes 's original doing! :) Yes, I'll update that to include action - thanks!

Reply:
strong emoji game all around!